src/exchange.js

```diff
@@ -6,18 +6,31 @@ var Helpers = require('./helpers');
 // https://gist.github.com/Zodiase/af44115098b20d69c531
 
 class Exchange {
-  constructor (delegate, TradeClass, QuoteClass, PaymentMediumClass) {
+  constructor (obj, delegate, api, TradeClass, QuoteClass, PaymentMediumClass) {
     assert(this.constructor !== Exchange, 'Abstract Class');
     assert(delegate, 'ExchangeDelegate required');
+    assert(api, 'API class required');
     assert(TradeClass, 'Trade class required');
     assert(QuoteClass, 'Quote class required');
     assert(QuoteClass, 'PaymentMethod class required');
     assert(QuoteClass.getQuote, 'Quote.getQuote missing');
     this._delegate = delegate;
+    this._api = api;
     this._trades = [];
     this._TradeClass = TradeClass;
     this._QuoteClass = QuoteClass;
     this._PaymentMediumClass = PaymentMediumClass;
+
+    this._trades = [];
+
+    if (obj.trades) {
+      for (let tradeObj of obj.trades) {
+        var trade = new TradeClass(tradeObj, this._api, delegate);
+        trade._getQuote = QuoteClass.getQuote; // Prevents circular dependency
```

Comment:
This is nothing new, although I agree it's ugly. `Trade` in Coinify has a method `btcExpected()` which uses `Quote.getQuote`. But because `quote.js` requires `trade.js`, `trade.js` can't require `quote.js`. That needs untangling, but maybe not as part of this refactor.